src/rgw/driver/rados/rgw_rados.cc

```diff
@@ -7922,10 +7923,9 @@ int RGWRados::bucket_index_clear_olh(const DoutPrefixProvider *dpp,
 			  [&](BucketShard *pbs) -> int {
 			    ObjectWriteOperation op;
 			    op.assert_exists(); // bucket index shard must exist
-			    auto& ref = pbs->bucket_obj.get_ref();
 			    cls_rgw_guard_bucket_resharding(op, -ERR_BUSY_RESHARDING);
 			    cls_rgw_clear_olh(op, key, olh_tag);
-                            return rgw_rados_operate(dpp, ref.pool.ioctx(), ref.obj.oid, &op, y);
+                            return rgw_rados_operate(dpp, ref.ioctx, ref.obj.oid, &op, y);
```

Comment:
adding back `auto& ref = pbs->bucket_obj` fixed the run-bucket-check failure. this had compiled without it, because `ref` shadowed another variable from `get_obj_head_ref()` above which was for the wrong object